src/core/send_buffer.c

```diff
@@ -228,17 +257,15 @@ QuicSendBufferConnectionAdjust(
         return; // Nothing to do.
     }
 
-    //
-    // If the current IdealBytes is close to limiting throughput, double it.
-    // Since we grow exponentially, this will not happen frequently.
+    const uint64_t NewIdealBytes =
+        QuicGetNextIdealBytes(Connection->CongestionControl.BytesInFlightMax);
+
     //
     // TODO: Currently, IdealBytes only grows and never shrinks. Add appropriate
     // shrinking logic.
     //
-    if (Connection->CongestionControl.BytesInFlightMax >
-        QUIC_IDEAL_SEND_BUFFER_THRESHOLD(Connection->SendBuffer.IdealBytes)) {
-        Connection->SendBuffer.IdealBytes =
-            min(2 * Connection->CongestionControl.BytesInFlightMax, QUIC_MAX_IDEAL_SEND_BUFFER_SIZE);
+    if (NewIdealBytes < Connection->SendBuffer.IdealBytes) {
```

Comment:
Is the direction wrong here? Or is this handling the shrinking case, and the comment is no longer appropriate?

Reply:
You're right. Wrong direction.